File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.c
int is_vowel(char v) {
    return strchr("aeiou", v) != NULL;
}

int maxVowels(char * s, int k){
    size_t vcount, maxv, l, r, slen;
    
    vcount = 0;
    for (size_t i = 0; i < k; i++)
        vcount += is_vowel(s[i]);
    
    maxv = vcount;
    l = 0;
    r = k;
    slen = strlen(s);
    while (r < slen) {
        if (maxv == k)
            return maxv;
        vcount = vcount - is_vowel(s[l++]) + is_vowel(s[r++]);
        maxv = (maxv >= vcount) ? maxv : vcount;
    }
    return maxv;
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.go
func maxVowels(s string, k int) int {
    vowelCount := 0
    for i := 0; i < k; i++ {
        if isVowel(string(s[i])) {
            vowelCount++
        }
    }
    maxVowels := vowelCount
    
    l, r := 0, k
    for r < len(s) {
        if maxVowels == k {
            return maxVowels
        }
        if isVowel(string(s[l])) {
            vowelCount--
        }
        if isVowel(string(s[r])) {
            vowelCount++
        }
        l++
        r++
        maxVowels = max(maxVowels, vowelCount)
    }
    return maxVowels
}

func isVowel(s string) bool {
    vowels := "aeiou"
    return strings.Contains(vowels, s)
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}